include/boost/geometry/algorithms/detail/overlay/discard_duplicate_turns.hpp

```diff
@@ -104,26 +104,31 @@ void discard_duplicate_start_turns(Turns& turns,
         return multi_and_ring_id_type{seg_id.multi_index, seg_id.ring_index};
     };
 
-    for (auto const& turn : turns)
+    for (auto& turn : turns)
     {
         // Any turn which "crosses" does not have a corresponding turn.
         // Also avoid comparing "start" with itself
         if (turn.method == method_crosses || turn.method == method_start)
         {
             continue;
         }
+        bool const is_touch = turn.method == method_touch;
         for (auto const& op : turn.operations)
         {
             auto it = start_turns_by_segment.find(adapt_id(op.seg_id));
-            if (it != start_turns_by_segment.end())
+            if (it == start_turns_by_segment.end())
             {
-                for (std::size_t const& i : it->second)
+                continue;
+            }
+            for (std::size_t const& i : it->second)
+            {
+                auto& start_turn = turns[i];
+                if (start_turn.cluster_id == turn.cluster_id
+                    && corresponding_turn(turn, start_turn, geometry0, geometry1))
                 {
-                    if (turns[i].cluster_id == turn.cluster_id
-                        && corresponding_turn(turn, turns[i], geometry0, geometry1))
-                    {
-                        turns[i].discarded = true;
-                    }
+                    // Discard the start turn, unless there is a touch before.
+                    // In that case the start is used and the touch is discarded.
+                    (is_touch ? turn : start_turn).discarded = true;
```

Comment:
▶️ This is the fix

Reply:
Similar to https://github.com/boostorg/geometry/pull/1341 this is again related to start turns and how to treat them in special cases.

Reply:
Indeed. It's quite related. I want to harmonize it at some point.
That one was about touch/touch_interior
This one is about touch/start ...

I want to build the map (segment -> turnindex) once for both options and then inspect.
But last week, this was not possible because the usecase is slightly different.

I'm trying to inspect it more and get related cases and refactor. But that will  be `1.88` I think.